test/unit/server_async_transaction.test.js

```diff
@@ -0,0 +1,122 @@
+                done();
+            })
+            .catch(function (err) {
+                done(err);
+            });
+    });
+    it("sends an async transaction and gets a Problem response", function (done) {
+        const response = {
+            type:   "transaction_submission_exception",
+            title:  "Transaction Submission Exception",
+            status: 500,
+            detail: "Received exception from stellar-core." +
+                "The `extras.error` field on this response contains further " +
+                "details.  Descriptions of each code can be found at: " +
+                "https://developers.stellar.org/api/errors/http-status-codes/horizon-specific/transaction-submission-async/transaction_submission_exception",
+            extras: {
+                envelope_xdr: "AAAAAIUAEW3jQt3+fbT6nCASA1/8RWdp9fJ2woxqPHZPQUH/AAAAZAEH/OgAAAAgAAAAAQAAAAAAAAAAAAAAAFyIDD0AAAAAAAAAAQAAAAAAAAADAAAAAAAAAAFCQVQAAAAAAEZK09vHmzOmEMoVWYtbbZcKv3ZOoo06ckzbhyDIFKfhAAAAAAExLQAAAAABAAAAAgAAAAAAAAAAAAAAAAAAAAFPQUH/AAAAQHk3Igj+JXqggsJBFl4mrzgACqxWpx87psxu5UHnSskbwRjHZz89NycCZmJL4gN5WN7twm+wK371K9XcRNDiBwQ=",
+                error: "There was an exception when submitting this transaction."
+            }
+        };
+
+        this.axiosMock
+            .expects("post")
+            .withArgs(
+                "https://horizon-live.stellar.org:1337/transactions_async",
+                `tx=${this.blob}`,
+            )
+            .returns(Promise.resolve({ data: response }));
+
+        this.server
+            .submitAsyncTransaction(this.transaction, { skipMemoRequiredCheck: true })
+            .then(function (res) {
+                expect(res.type).to.equal("transaction_submission_exception")
+                expect(res.title).to.equal("Transaction Submission Exception")
+                expect(res.status).to.equal(500)
+                expect(res.detail).to.equal("Received exception from stellar-core." +
+                    "The `extras.error` field on this response contains further " +
+                    "details.  Descriptions of each code can be found at: " +
+                    "https://developers.stellar.org/api/errors/http-status-codes/horizon-specific/transaction-submission-async/transaction_submission_exception")
+                expect(res.extras.envelope_xdr).to.equal("AAAAAIUAEW3jQt3+fbT6nCASA1/8RWdp9fJ2woxqPHZPQUH/AAAAZAEH/OgAAAAgAAAAAQAAAAAAAAAAAAAAAFyIDD0AAAAAAAAAAQAAAAAAAAADAAAAAAAAAAFCQVQAAAAAAEZK09vHmzOmEMoVWYtbbZcKv3ZOoo06ckzbhyDIFKfhAAAAAAExLQAAAAABAAAAAgAAAAAAAAAAAAAAAAAAAAFPQUH/AAAAQHk3Igj+JXqggsJBFl4mrzgACqxWpx87psxu5UHnSskbwRjHZz89NycCZmJL4gN5WN7twm+wK371K9XcRNDiBwQ=")
+                expect(res.extras.error).to.equal("There was an exception when submitting this transaction.")
+                done();
+            })
+            .catch(function (err) {
+                done(err);
+            });
```

Comment:
same suggestion here re: shortening:
```suggestion
            .catch((err) => done(err));
```